### PR TITLE
skypilot: init at 0.6.0

### PR DESCRIPTION
**Pull request by seanrmurphy**
## Description of changes

- Add python package skypilot. This generates an executable called `sky`.
- skypilot git repo: https://github.com/skypilot-org/skypilot
-  v0.6.0 release notes: https://github.com/skypilot-org/skypilot/releases/tag/v0.6.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by aos — COMMENTED** [review]
You should add a `meta` section: https://nixos.org/manual/nixpkgs/stable/#chap-meta

**Comment by seanrmurphy** [comment]
> You should add a `meta` section: https://nixos.org/manual/nixpkgs/stable/#chap-meta

@aos - thanks for the feedback - apols for forgetting this - have added it now.

File: pkgs/by-name/sk/skypilot/package.nix [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  meta = {
+    description = "Run LLMs and AI on Any Cloud";
+    longDescription = ''
+      SkyPilot is a framework for running LLMs, AI, and batch jobs on any
+      cloud, offering maximum cost savings, highest GPU availability, and
+      managed execution.
+    '';
+    homepage = "https://github.com/skypilot-org/skypilot";
+    license = lib.licenses.asl20;
+    maintainers = with lib.bmaintainers; [ seanrmurphy ];
```
Comment (aos) [inline pkgs/by-name/sk/skypilot/package.nix:68]:
```suggestion
    maintainers = with lib.maintainers; [ seanrmurphy ];
```

File: pkgs/by-name/sk/skypilot/package.nix [hunk 2]
```diff
@@ -0,0 +1,70 @@
+{
+  lib,
+  pkgs,
+  stdenv,
+  fetchFromGitHub,
+  python3Packages
+}:
+
+python3Packages.buildPythonPackage rec {
```
Comment (aos) [inline pkgs/by-name/sk/skypilot/package.nix:9]: I haven't looked too deeply at the source, but it seems like the package you're building is only used purely as a binary, in which case this should be `buildPythonApplication` instead. If it's the case that you plan on importing the module, then you can keep as is.

See: https://nixos.org/manual/nixpkgs/unstable/#buildpythonapplication-function

Comment (seanrmurphy) [reply pkgs/by-name/sk/skypilot/package.nix:9]: Good point and thanks for the feedback @aos - in its current form, it really is an application.

I will modify as you suggest.

Comment (seanrmurphy) [reply pkgs/by-name/sk/skypilot/package.nix]: That's done now.

**Review by aos — APPROVED** [review]
Result of `nixpkgs-review pr 324602` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>skypilot</li>
    <li>skypilot.dist</li>
  </ul>
</details>

**Comment by seanrmurphy** [comment]
I removed the commit in this PR which adds me to the maintainer list as I've been added a few hours ago via #322573.

**Comment by seanrmurphy** [comment]
Noticed that I had left out `mainProgram` in the metadata - added it here as the name differs from the package name.

File: pkgs/by-name/sk/skypilot/package.nix [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    psutil
+    python-dotenv
+    pyyaml # pyyaml >3.13 != 5.4 is required; NixOS provides 6.0.1
+    pulp
+    requests
+    rich
+    tabulate
+    # From source (skypilot v0.6.0)
+    # Light weight requirement, can be replaced with "typing" once
+    # we deprecate Python 3.7 (this will take a while).
```
Comment (SuperSandro2000) [inline pkgs/by-name/sk/skypilot/package.nix:46]: Please drop all of those comments. They will be insanely quick out of date and we generally do not care about those precise details unless they have some importance.

Comment (seanrmurphy) [reply pkgs/by-name/sk/skypilot/package.nix]: Thanks @SuperSandro2000 - I have removed them but added a comment indicating that when updating, the `setup.py` file should be checked to ensure package versions are ok. Hope that's ok.

**Pull request merged**